### ISSUE TITLE
Black Bar Appears at Top Default Theme pages in Safari Light Mode

### ISSUE THREAD

**Issue by valgaze** [vuejs/vitepress#3223]
TL;DR: In Safari/Webkit 15.2, `rgba(0,0,0,0)` is rendered as a transparent black and the top nav-bar "frost" gradient is dark/looks bad on light-mode accordingly

----

### Describe the bug

When accessing a Vitepress page in Safari's v15.2 in light mode, a noticeable black bar appears at the top of the page when scrolling. It seems likely that this issue is related to a CSS or browser quirk specific to Safari since it looks beautiful in chrome and non-webkit browsers

On chrome, the scroll shadow looks like this ✅:

![shadow_good_chrome](https://github.com/vuejs/vitepress/assets/1396559/dce18077-77e4-49e0-8c15-0e90c7029fa9)

On safari, the scroll shadow looks like this ❌:

![shadow_breaking_safari](https://github.com/vuejs/vitepress/assets/1396559/0e511c4b-0544-4e8f-b966-da9e5105f7e0)

Looks frost on Firefox & friends:

<img width="1552" alt="image" src="https://github.com/vuejs/vitepress/assets/1396559/8f0d7742-ee44-4f0a-8cea-6e090651df15">

### Reproduction

Open any Vitepress with scrolling content in Safari in light mode: https://vitepress.dev/guide/getting-started

### Expected behavior

Expect: light mode in Safari should have a "frost"/white shadow rather than the dark bar 

### System Info

https://github.com/vuejs/vitepress/blob/85bd7e845ab37862d68d2506f0b693dafc236a7a/src/client/theme-default/components/VPNavBar.vue#L222

```Text
n/a
```

### Additional context

Good test site for Safari: https://vitepress.dev/guide/getting-started

<img width="1552" alt="image" src="https://github.com/vuejs/vitepress/assets/1396559/50268366-66c8-4e68-a7aa-0af204b4794c">

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Which safari version are you on. Can't seem to reproduce this on v17.1 and TP

**Comment by valgaze** [comment]
@brc-dd Whoops sorry about leaving crucial information like that out-- v15.2 (haven't upgraded my OS yet)


<img width="396" alt="image" src="https://github.com/vuejs/vitepress/assets/1396559/1a15e418-1b5c-4616-b835-4b917defaab5"> 15.2 (17612.3.6.1.6)

**Comment by valgaze** [comment]
It sounds like this associated w/ how Wk handles `transparent` in `background: linear-gradient(var(--vp-c-bg), transparent 70%);`

- https://stackoverflow.com/questions/38391457/linear-gradient-to-transparent-bug-in-latest-safari
- https://ambientimpact.com/web/snippets/safari-bug-with-gradients-that-fade-to-transparent

It sounds like the fix involves something like this:

The 2nd RGB color below which takes place of transparent would need to be exact color but a fully transparent version

```
.VPNavBar.has-sidebar .curtain::before {
  display: block;
  width: 100%;
  height: 32px;
  background: linear-gradient(var(--vp-c-bg), rgba(100,100,100, 0) 70%);
  content: "";
}
```

**Comment by valgaze** [comment]
Ex this is a rough rgb(255,255,255,0) (note alpha transparency) that looks reasonable on Safari (but needs to change in dark-mode):

```
  .VPNavBar.has-sidebar .curtain::before {
    display: block;
    width: 100%;
    height: 32px;
    background: linear-gradient(var(--vp-c-bg), rgba(255, 255, 255, 0) 70%);
    content: "";
  }
  ```

**Comment by valgaze** [comment]
Bottom line here:

- The `transparent` line here is causing trouble: https://github.com/vuejs/vitepress/blob/85bd7e845ab37862d68d2506f0b693dafc236a7a/src/client/theme-default/components/VPNavBar.vue#L222

- `transparent` is shorthand for rgba(0,0,0,0) (**[per w3](https://www.w3.org/TR/css-color-3/#transparent)**)

- In safari/wk `transparent` is rendered as transparent but black

Sass is a potential solution out of this with a processing step

**Comment by brc-dd** [comment]
That curtain is removed in #3359. It should be fine now.
